packages/react-checkbox/src/components/Checkbox/Checkbox.types.ts

```diff
@@ -68,13 +68,8 @@ export type CheckboxSlots = {
 /**
  * Checkbox Props
  */
-export type CheckboxProps = Omit<ComponentProps<CheckboxSlots>, 'defaultChecked'> &
+export type CheckboxProps = Omit<ComponentProps<CheckboxSlots, 'input'>, 'size' | 'checked' | 'defaultChecked'> &
```

Comment:
I think for consistency with other fields and to eliminate the native prop overlap issue, we should consider renaming `size` to `fieldSize`. That's what I have in Input and plan to do in other input components for consistency.

Reply:
Could be a good idea, but that's out of scope of this PR. Could you make a PR that makes the change for all input components?

Reply:
Made an issue for now, #20733